project_forecast_line_priority/models/project_task.py

```diff
@@ -0,0 +1,66 @@
+# Copyright 2024 Therp BV
+# License AGPL-3.0 or later (https://www.gnu.org/licenses/agpl).
+from datetime import timedelta
+
+from odoo import fields, models
+
+
+class ProjectTask(models.Model):
+    _inherit = "project.task"
+
+    def write(self, vals):
+        """Set forecast_date_planned_end based on priority"""
+        # If user sets end date manually
+        if "forecast_date_planned_end" in vals or "date_deadline" in vals:
+            return super().write(vals)
+        if "priority" not in vals:
+            return super().write(vals)
+        priority = vals.get("priority", -5)
+        for this in self:
+            # date deadline is set, so ignore this one
+            if this.date_deadline:
+                continue
+            # if priority is not set, do nothing
+            if int(priority) < 0:
+                continue
+            forecast_date_planned_end = self._get_forecast_date_planned(
+                priority=priority
+            )
+            if not forecast_date_planned_end:
+                continue
+            vals["forecast_date_planned_end"] = forecast_date_planned_end
+        return super().write(vals)
+
+    def _update_forecast_lines(self):
+        """Override cron method and inject forecast date recomputation"""
+        for task in self:
+            forecast_date_planned_end = task._get_forecast_date_planned()
+            if not forecast_date_planned_end:
+                continue
+            task.forecast_date_planned_end = forecast_date_planned_end
+        return super()._update_forecast_lines()
+
+    def _get_forecast_date_planned(self, priority=None):
+        """Update forecast date end based on priority"""
+        self.ensure_one()
+        if self.date_deadline:
+            return False
+        config_model = self.env["ir.config_parameter"]
+        priority = priority or self.priority
+        selection = config_model.get_param(
+            "project_forecast_line_priority.priority_%s_selection" % priority
+        )
+        if selection == "none":
+            return False
+        if selection == "delta":
+            return fields.Date.today() + timedelta(
+                days=int(
+                    config_model.get_param(
+                        "project_forecast_line_priority.priority_%s_delta" % priority
+                    )
+                )
+            )
+        if selection == "date":
+            return config_model.get_param(
+                "project_forecast_line_priority.priority_%s_date" % priority
+            )
```

Comment:
What is selection anything else? Either return False or maybe throw an Exception as something that should not happen. I think returning False will be enough.